lib/java/collections.nit

```diff
@@ -46,6 +46,48 @@ extern class AbstractJavaArray[E: Object]
 	redef fun reverse_iterator do return new JavaArrayReverseIterator[E](self)
 end
 
+#Java primitive array `int[]`
+extern class JavaIntArray in "Java" `{ int[] `}
+	super AbstractJavaArray[Int]
+
+	# Get a new array of the given `size`
+	new (size: Int) in "Java" `{ return new int[(int)size]; `}
+
+	redef fun [](i) in "Java" `{ return recv[(int)i]; `}
+
+	redef fun []=(i, e) in "Java" `{ recv[(int)i] = (int)e; `}
+
+	redef fun length in "Java" `{ return recv.length; `}
+end
+
+#Java primitive array `short[]`
+extern class JavaShortArray in "Java" `{ short[] `}
+	super AbstractJavaArray[Int]
+
+	# Get a new array of the given `size`
+        new (size: Int) in "Java" `{ return new short[(int)size]; `}
+
+        redef fun [](i) in "Java" `{ return (short)recv[(int)i]; `}
+
+        redef fun []=(i, e) in "Java" `{ recv[(int)i] = (short)e; `}
+
+        redef fun length in "Java" `{ return recv.length; `}
+end
+
+#Java primitive array `long[]`
+extern class JavaLongArray in "Java" `{ long[] `}
+	super AbstractJavaArray[Int]
+
+	# Get a new array of the given `size`
+        new (size: Int) in "Java" `{ return new long[(int)size]; `}
+
+        redef fun [](i) in "Java" `{ return (long)recv[(int)i]; `}
+
+        redef fun []=(i, e) in "Java" `{ recv[(int)i] = (long)e; `}
```

Comment:
The `(long)` casts should be superflous in this class, a Nit `Int` translates to a Java `long` (if I'm not mistaken).